code/go/0chain.net/chaincore/state/state.go

```diff
@@ -1,17 +1,27 @@
 package state
 
 import (
-	"0chain.net/core/encryption"
-	"0chain.net/core/logging"
-	"0chain.net/core/util"
+	"math"
 	"bytes"
 	"encoding/binary"
 	"encoding/hex"
+
+	"0chain.net/core/encryption"
+	"0chain.net/core/logging"
+	"0chain.net/core/util"
 )
 
 //Balance - any quantity that is represented as an integer in the lowest denomination
 type Balance int64
 
+func (b Balance) Min(v Balance) Balance {
```

Comment:
It will be kept!
As it should be used in similar cases.
The legacy code is lack of level of utility functions that makes it "unreadable" and "duplicate".

Reply:
It is the block copying that makes it feel duplicated. It is being badly written that makes it a chore to read. I would agree we should be looking to merge common code. However, making new utility functions for the sake of it is not a solution.
```go
   math.Min(b,v)
```
or 
```go
  b.Min(v)
```
Is the second that much clear than the first that we need to be making extra changes in unrelated files?

Reply:
Fix an obvious errors in your code example first. It will not compile.
Then we continue...